### PR TITLE
Check multiple roles in blade with hasanyrole and hasallrole

### PR DESCRIPTION
**Pull request by coderkk**
Allow blade to check multiple roles

hasanyrole
```
@hasanyrole('Admin|Editor')
I have admin or Editor roles
@endhasanyrole
```

hasallrole
```
@hasallrole('Admin|Editor')
I have admin and Editor roles
@endhasanyrole
```

### CODE REVIEW

File: tests/BladeTest.php [hunk 1]
```diff
@@ -142,6 +142,16 @@ public function the_hasanyrole_directive_will_evaluate_true_when_the_logged_in_u
     }
 
     /** @test */
+    public function the_hasanyrole_directive_will_evaluate_true_when_the_logged_in_user_does_have_some_of_the_required_roles_use_pipe_for_the_given_guard()
```
Comment (freekmurze) [inline tests/BladeTest.php]: This method name is a bit too long for my taste. What do you think about losing the `use_pipe_for_the_given_guard` part? (same remarks for the other test you added)

File: src/PermissionServiceProvider.php [hunk 2]
```diff
@@ -69,6 +69,16 @@ protected function registerBladeExtensions()
 
             $bladeCompiler->directive('hasanyrole', function ($arguments) {
                 list($roles, $guard) = explode(',', $arguments.',');
+                $roles = trim($roles);
```
Comment (freekmurze) [inline src/PermissionServiceProvider.php]: I've not followed it in detail, but my gut says that this could / should be solved in a much simpler way. Since the tests prove that this work, it should be relatively save to attempt to refactor this.

Also: this logic is being repeated for the other blade directive. If it can't be made simpler, create a new function so that logic appears only once in the codebase.

File: tests/BladeTest.php [hunk 3]
```diff
@@ -142,6 +142,16 @@ public function the_hasanyrole_directive_will_evaluate_true_when_the_logged_in_u
...
+    {
+        $guard = 'admin';
+
+        auth('admin')->setUser($this->getSuperAdmin());
+
+        $this->assertEquals('does have some of the roles', $this->renderView('guardHasAnyRolePipe', compact('guard')));
```
Comment (freekmurze) [inline tests/BladeTest.php]: Could you also add a test that checks if `does not have any of the given roles` is returned when a user has none of the requested roles?

File: tests/BladeTest.php [hunk 4]
```diff
@@ -184,6 +194,22 @@ public function the_hasallroles_directive_will_evaluate_true_when_the_logged_in_
...
+
+        $admin = $this->getSuperAdmin();
+
+        $admin->assignRole('moderator');
+
+        $this->refreshTestAdmin();
+
+        auth('admin')->setUser($admin);
+
+        $this->assertEquals('does have all of the given roles', $this->renderView('guardHasAllRolesPipe', compact('guard')));
```
Comment (freekmurze) [inline tests/BladeTest.php]: Could you add a tests the checks if `does not have all of the given roles` when the user does not have all the given roles?

File: tests/BladeTest.php (hunk 1 above)
Comment (coderkk) [inline tests/BladeTest.php]: OK, I will change it

File: tests/BladeTest.php (hunk 3 above)
Comment (coderkk) [inline tests/BladeTest.php]: OK, I will add another test for not match

File: tests/BladeTest.php (hunk 4 above)
Comment (coderkk) [inline tests/BladeTest.php]: OK, I will add another test for not match

File: src/PermissionServiceProvider.php (hunk 2 above)
Comment (coderkk) [inline src/PermissionServiceProvider.php]: Sure, no problem

**Comment by coderkk** [comment]
Hi,  I have done the request change.

**Comment by freekmurze** [comment]
Hi, thank you. I'm pulling this in an work a little bit on it myself.
